apps/remote_control/lib/lexical/remote_control/code_mod/extract_function.ex

```diff
@@ -0,0 +1,245 @@
+defmodule Lexical.RemoteControl.CodeMod.ExtractFunction do
```

Comment:
This feels like it should be a code action handler, and Code Action handlers have the following behaviour
```elixir
  @callback actions(Document.t(), Range.t(), [Diagnostic.t()]) :: [CodeAction.t()]
  @callback kinds() :: [CodeAction.code_action_kind()]
```

Reply:
You should be making use of the `Lexical.Ast` to build Ast here. there are a lot of helper functions in there. I'm wondering if it makes sense to pull out the range first via `Document.fragment(document, range)`, turn _that_ into AST, make it fairly generic and then find all generic usages of that AST and replace them with the function you've created.